workers/data_refinery_workers/processors/illumina.R

```diff
@@ -0,0 +1,447 @@
+  #             help="GSE", metavar="character"),
+  make_option(c("-p", "--probeId"), type="character", default="PROBE_ID", 
+              help="Probe ID", metavar="character"),
+  make_option(c("-e", "--expression"), type="character", default=".AVG_Signal", 
+              help="expression", metavar="character"),
+  make_option(c("-d", "--detection"), type="character", default="Detection Pval", 
+              help="Detection Pval", metavar="character"),
+  make_option(c("-l", "--platform"), type="character", default="illuminaHumanv4", 
+              help="Platform", metavar="character"),
+  make_option(c("-c", "--cores"), type="character", default="1", 
+              help="Number of cores", metavar="character"),
+  make_option(c("-i", "--inputFile"), type="character", default="", 
+              help="inputFile", metavar="character"),
+  make_option(c("-o", "--outputFile"), type="character", default="", 
+              help="outputFile", metavar="character")
+); 
+
+opt_parser = OptionParser(option_list=option_list);
+opt = parse_args(opt_parser);
+
+# gseID <- opt$gse
+probeIDColumn <- opt$probeId
+exprColumns <- strsplit(opt$expression, ",")
+exprColumns <- unlist(lapply(exprColumns, as.integer))
+detectionPValueColumnPattern <- opt$detection
+platform <- opt$platform
+numCores <- strtoi(opt$cores)
+filePath <- opt$inputFile
+outFilePath <- opt$outputFile
+
+library(limma)
+library(oligo)
+library(doParallel)
+library(data.table)
+library(paste(platform, ".db", sep=""), character.only=TRUE)
+
+# Read the data file
+message("Reading data file...")
+suppressWarnings(data <- fread(filePath, stringsAsFactors=FALSE, sep="\t", header=TRUE, autostart=10, data.table=FALSE, check.names=FALSE, fill=TRUE, na.strings="", showProgress=FALSE))
+#suppressWarnings(data <- read.table(filePath, header=TRUE, fill=TRUE))
+
+# Check input paramters and parse out data we need
+if (probeIDColumn == ""){
+  message("Got no probID column...")
+  probeIDColumn <- "PROBE_ID"
+}
+
+if (!(probeIDColumn %in% colnames(data)))
+{
+  stop(paste0("No ", probeIDColumn, " column in the data file."))
+}
+
+probeIDs <- data[,probeIDColumn]
+
+
+if (length(exprColumns) == 0)
+  stop(paste0("No columns match this pattern: ", exprColumnPattern))
+
+exprData <- as.matrix(data[,exprColumns,drop=FALSE])
+rownames(exprData) <- probeIDs
+
+if (detectionPValueColumnPattern == "")
+  detectionPValueColumnPattern <- "Detection Pval"
+pValueColumns <- grep(detectionPValueColumnPattern, colnames(data), ignore.case=TRUE)
+
+if (length(pValueColumns) == 0)
+  stop(paste0("No columns match this pattern: ", detectionPValueColumnPattern))
+
+if (length(exprColumns) != length(pValueColumns))
+  stop(paste0("The number of expression columns [", length(exprColumns), "] did not match the number of p-value columns [", length(pValueColumns), "]."))
+
+pValueData <- as.matrix(data[,pValueColumns,drop=FALSE])
+rownames(pValueData) <- probeIDs
+
+# Extract platform-specific data
```

Comment:
This only supports 2 human platforms, Illumina HumanWG-6 v2.0 (i.e., [`GPL13376`](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL13376)) and Illumina HumanHT-12 V4.0 (i.e., [`GPL10558`](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL10558)), using the relevant bioconductor packages at the moment.

We'll want to support a broader set of BeadChips. Specifically, there are bioconductor packages for the following additional platforms:

* [`illuminaHumanv1.db`](https://doi.org/doi:10.18129/B9.bioc.illuminaHumanv1.db)
* [`illuminaHumanv3.db`](https://doi.org/doi:10.18129/B9.bioc.illuminaHumanv3.db)
* [`illuminaMousev1.db`](https://doi.org/doi:10.18129/B9.bioc.illuminaMousev1.db)
* [`illuminaMousev1p1.db`](https://doi.org/doi:10.18129/B9.bioc.illuminaMousev1p1.db)
* [`illuminaMousev2.db`](https://doi.org/doi:10.18129/B9.bioc.illuminaMousev2.db)
* [`illuminaRatv1.db`](https://doi.org/doi:10.18129/B9.bioc.illuminaRatv1.db)

You also might consider getting the probe objects like so: 
```R
probeSequenceRef <- lazyeval::lazy_eval(paste0(platform, ".db::", platform, "PROBESEQUENCE"))
probeQualityRef <- lazyeval::lazy_eval(paste0(platform, ".db::", platform, "PROBEQUALITY"))
probeGeneRef <-  lazyeval::lazy_eval(paste0(platform, ".db::", platform, "ENSEMBLREANNOTATED"))
```

☝️ would also eliminate the need for loading the annotation library (line 356).